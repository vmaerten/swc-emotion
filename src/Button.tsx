import styled from "@emotion/styled";



export const Child = styled.div`
  color: red;
`

export const Parent = styled.div`
  ${Child} {
    color: green;
  }
`

export const Button = styled.button`
  font-weight: bold;
  padding: 9px 11px;
  cursor: pointer;
  border: 1px solid transparent;
  border-radius: 4px;
`
