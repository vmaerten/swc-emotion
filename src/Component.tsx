import styled from "@emotion/styled";

const OneComponent = (props: any) => {
  return (
      <div>
        just a div
      </div>
  )
}

export const Child = styled(OneComponent)`
  background: #535bf2;
  color: red;
`

export const Parent = styled.div`
  right: 34px;
  ${Child} ~ & {
    left: 36px;
  }
  div {
    top : 42px;
  }
`
