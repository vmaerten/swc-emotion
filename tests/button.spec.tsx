import React from 'react'
import renderer from 'react-test-renderer'
import {Button, Child, Parent} from "../src/Button";
import {test, expect} from 'vitest'


test('Button renders correctly', () => {
    expect(
        renderer.create(  <div>
            <Parent>
                <Child>Green because I am inside a Parent</Child>
            </Parent>
            <Child>Red because I am not inside a Parent</Child>
        </div>).toJSON()
).toMatchSnapshot()
})