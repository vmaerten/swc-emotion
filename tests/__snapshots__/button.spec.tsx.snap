// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button renders correctly 1`] = `
<div>
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-1uhjhvs .ehnzxxk0{color:green;}",
      }
    }
    data-emotion="css 1uhjhvs"
  />
  <div
    className="css-1uhjhvs ehnzxxk1"
  >
    <style
      dangerouslySetInnerHTML={
        {
          "__html": ".css-tokvmb{color:red;}",
        }
      }
      data-emotion="css tokvmb"
    />
    <div
      className="css-tokvmb ehnzxxk0"
    >
      Green because I am inside a Parent
    </div>
  </div>
  <style
    dangerouslySetInnerHTML={
      {
        "__html": ".css-tokvmb{color:red;}",
      }
    }
    data-emotion="css tokvmb"
  />
  <div
    className="css-tokvmb ehnzxxk0"
  >
    Red because I am not inside a Parent
  </div>
</div>
`;
